File: level_2/hard3_find_num_subtree_structure/langs/compiler.cpp
int main()
{
    int N, E, R;
    int output_subtree_count=0;
    cin >> R;
    cin >> N;
    cin >> E;

    unordered_map<int, vector<int>> tree;

    for (int i = 0; i < E; ++i) 
    {
        int node1, node2;
        cin >> node1 >> node2;
        tree[node1].push_back(node2);
        tree[node2].push_back(node1);
    }
    
    find_num_subtree_structure(N, R, tree, &output_subtree_count);
    
    cout << output_subtree_count;

    return 0;
}

File: level_2/hard3_find_num_subtree_structure/langs/compiler.py
def main():
    R, N, E = map(int, input().split())
    tree = create_tree_array(N + 1)
    for _ in range(E):
        node1, node2 = map(int, input().split())
        n1 = Node(node2)
        n1.next = tree[node1].neighbors
        tree[node1].neighbors = n1
        n2 = Node(node1)
        n2.next = tree[node2].neighbors
        tree[node2].neighbors = n2
    
    output_subtree_count = [0]
    find_num_subtree_structure(N, R, tree, output_subtree_count)

    print(output_subtree_count[0])

if __name__ == "__main__":
    main()
